ena-submission/src/ena_deposition/create_assembly.py

```diff
@@ -429,6 +429,90 @@ def can_be_revised(config: Config, db_config: SimpleConnectionPool, entry: dict[
     return True
 
 
+def no_change(config: Config, db_config: SimpleConnectionPool, entry: dict[str, Any]) -> bool:
+    """
+    Check if no change in sequence or flatfile metadata has occurred since last version.
+    """
+    version_to_revise = last_version(db_config, entry)
+    last_version_data = find_conditions_in_db(
+        db_config,
+        table_name=TableName.SUBMISSION_TABLE,
+        conditions={"accession": entry["accession"], "version": version_to_revise},
+    )
+    if len(last_version_data) == 0:
+        error_msg = f"Last version {version_to_revise} not found in submission_table"
+        raise RuntimeError(error_msg)
+
+    if (
+        entry["unaligned_nucleotide_sequences"]
+        != last_version_data[0]["unaligned_nucleotide_sequences"]
+    ):
+        logger.debug(
+            f"Unaligned nucleotide sequences have changed for {entry['accession']}, "
+            f"from {version_to_revise} to {entry['version']} - should be revised"
+        )
+        return False
+
+    fields = [
+        config.embl_property_fields.country_property,
+        config.embl_property_fields.collection_date_property,
+        config.embl_property_fields.authors_property,
+        *config.embl_property_fields.admin_level_properties,
+    ]
+    for field in fields:
+        last_entry = last_version_data[0]["metadata"].get(field)
+        new_entry = entry["metadata"].get(field)
+        if last_entry != new_entry:
+            logger.debug(
+                f"Field {field} has changed from {last_entry} to {new_entry} "
+                f"for {entry['accession']}"
+            )
+            return False
+    logger.debug(
+        f"No changes detected for {entry['accession']} from version {version_to_revise} "
+        f"to {entry['version']}"
+    )
+    return True
+
+
+def update_assembly_results_with_latest_version(
```

Comment:
This could use the retry refactor changes (once merged)

Reply:
done :-)